controllers/discovery/searchUser.js

```diff
@@ -18,15 +18,32 @@ const SearchUser = async (req, res) => {
     });
 
   try {
+    let user_topics = await sequelize.query(
+      `select tp.topic_id from topics as tp
+      left join user_topics as utp on tp.topic_id = utp.topic_id
+      where utp.user_id = :userId`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          userId
+        }
+      }
+    );
+    let topicIds = user_topics.map((topic) => topic.topic_id);
+    const similarTopicQuery =
+      topicIds.length > 0
+        ? `ARRAY( select name from topics as tp
+          left join user_topics as utp on tp.topic_id = utp.topic_id
+          where utp.user_id = "User".user_id and tp.topic_id in (:topicIds) limit 3
+        )`
+        : 'ARRAY[]::text[]';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code block constructs a part of the SQL query based on the length of `topicIds`. This could potentially lead to SQL injection if `topicIds` contains malicious values. It's better to always use parameterized queries or ORM methods to construct queries.

```diff
-    const similarTopicQuery =
-      topicIds.length > 0
-        ? `ARRAY( select name from topics as tp
-          left join user_topics as utp on tp.topic_id = utp.topic_id
-          where utp.user_id = "User".user_id and tp.topic_id in (:topicIds) limit 3
-        )`
-        : 'ARRAY[]::text[]';
+    const similarTopicQuery = sequelize.literal(`ARRAY(
+      SELECT name FROM topics AS tp
+      LEFT JOIN user_topics AS utp ON tp.topic_id = utp.topic_id
+      WHERE utp.user_id = "User".user_id AND tp.topic_id IN (:topicIds)
+      LIMIT 3
+    )`);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->